### PR TITLE
Separate scorecard cache + 2h TTL

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/scorecard/backend.dart [hunk 1]
```diff
@@ -124,9 +124,11 @@ class ScoreCardBackend {
         return null;
       }
     }
-    final cacheEntry = onlyCurrent || showSandboxedOutput
+    final cacheEntry = onlyCurrent
         ? null
-        : cache.scoreCardData(packageName, packageVersion);
+        : (showSandboxedOutput
+            ? cache.scoreCardData2(packageName, packageVersion)
```
Comment (jonasfj) [inline app/lib/scorecard/backend.dart:130]: I think we need to also store something in `scoreCardData2`.

Also we probably need purge from within the task backend.

Comment (isoos) [reply app/lib/scorecard/backend.dart:130]: The store is automatic in this method.
I've added calls for the purge.

Comment (jonasfj) [reply app/lib/scorecard/backend.dart:130]: > The store is automatic in this method.

The `    if (showSandboxedOutput) {` will return directly, without storing something in the cache.

Comment (isoos) [reply app/lib/scorecard/backend.dart:130]: fixed, PTAL

**Review by jonasfj — APPROVED** [review]

File: app/lib/scorecard/backend.dart [hunk 2]
```diff
@@ -166,6 +168,11 @@ class ScoreCardBackend {
         ),
         panaReport: PanaReport.fromSummary(summary, packageStatus: status),
       );
+      // only full cards will be stored in cache
+      if (cacheEntry != null && data.isCurrent && data.hasAllReports) {
```
Comment (jonasfj) [inline app/lib/scorecard/backend.dart:172]: We purge when updating, so why not cache incomplete stuff?

**Pull request merged**